To: vim_dev@googlegroups.com
Subject: Patch 7.3.179
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.179
Problem:    C-indent doesn't handle colon in string correctly.
Solution:   Skip the string. (Lech Lorens)
Files:	    src/misc1.c, src/testdir/test3.in, src/testdir/test3.ok


*** ../vim-7.3.178/src/misc1.c	2011-05-10 11:39:13.000000000 +0200
--- src/misc1.c	2011-05-10 11:50:14.000000000 +0200
***************
*** 5801,5807 ****
  		continue;
  	}
  
! 	if (s[0] == ':')
  	{
  	    if (s[1] == ':')
  	    {
--- 5801,5809 ----
  		continue;
  	}
  
! 	if (s[0] == '"')
! 	    s = skip_string(s) + 1;
! 	else if (s[0] == ':')
  	{
  	    if (s[1] == ':')
  	    {
*** ../vim-7.3.178/src/testdir/test3.in	2011-05-10 11:39:13.000000000 +0200
--- src/testdir/test3.in	2011-05-10 11:53:02.000000000 +0200
***************
*** 1360,1365 ****
--- 1360,1378 ----
  }
  
  STARTTEST
+ :set cino&
+ 2kdd=][
+ ENDTEST
+ 
+ void func(void)
+ {
+ 	cout << "a"
+ 	<< "b"
+ 	<< ") :"
+ 	<< "c";
+ }
+ 
+ STARTTEST
  :g/^STARTTEST/.,/^ENDTEST/d
  :1;/start of AUTO/,$wq! test.out
  ENDTEST
*** ../vim-7.3.178/src/testdir/test3.ok	2011-05-10 11:39:13.000000000 +0200
--- src/testdir/test3.ok	2011-05-10 11:50:14.000000000 +0200
***************
*** 1216,1218 ****
--- 1216,1227 ----
  	printf("Foo!\n");
  }
  
+ 
+ void func(void)
+ {
+ 	cout << "a"
+ 		<< "b"
+ 		<< ") :"
+ 		<< "c";
+ }
+ 
*** ../vim-7.3.178/src/version.c	2011-05-10 11:39:13.000000000 +0200
--- src/version.c	2011-05-10 11:53:36.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     179,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
71. You wonder how people walk

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
